File: web/src/services/auth.service.ts
import { request } from './apiClient';
import type { UserRole } from '../types';

export interface AuthUser {
  user_id: string;
  full_name: string;
  email: string;
  phone: string | null;
  role_id: string;
  status: 'ACTIVE' | 'INACTIVE';
  created_at: string;
  updated_at: string;
  role: { role_id: string; role_name: UserRole; description: string };
}

export interface LoginResult {
  token: string;
  user: AuthUser;
}

export function login(email: string, password: string): Promise<LoginResult> {
  return request<LoginResult>({ method: 'POST', url: '/auth/login', data: { email, password } });
}

export function getMe(): Promise<AuthUser> {
  return request<AuthUser>({ method: 'GET', url: '/auth/me' });
}

File: mobile/lib/services/auth_service.dart
import '../models/user.dart';
import 'api_client.dart';

class LoginResult {
  final String token;
  final AuthUser user;
  LoginResult({required this.token, required this.user});
}

/// Gọi đúng 2 endpoint thật của Backend (xem `web/src/services/auth.service.ts`
/// — cùng 1 bộ API dùng chung cho Web + Mobile).
class AuthService {
  AuthService._();
  static final instance = AuthService._();

  Future<LoginResult> login(String email, String password) async {
    final data = await ApiClient.instance.post<Map<String, dynamic>>(
      '/auth/login',
      data: {'email': email, 'password': password},
    );
    return LoginResult(
      token: data['token'] as String,
      user: AuthUser.fromJson(data['user'] as Map<String, dynamic>),
    );
  }

  Future<AuthUser> getMe() async {
    final data = await ApiClient.instance.get<Map<String, dynamic>>('/auth/me');
    return AuthUser.fromJson(data);
  }
}
